barretenberg/cpp/src/barretenberg/stdlib/primitives/bigfield/bigfield.hpp

```diff
@@ -300,6 +300,10 @@ template <typename Builder, typename T> class bigfield {
         result.set_free_witness_tag();
         return result;
     }
+    // Disallow from_witness for non-bb::fr types to prevent implicit conversions (specifically, using indices rather
+    // than values)
+    template <typename OT> static bigfield from_witness(Builder* ctx, const OT& input) = delete;
+    template <typename OT> static bigfield from_witness(Builder* ctx, const OT input) = delete;
```

Comment:
Not critical but I'm fairly certain that the `OT&` case captures everything (and more) that the `OT` case captures so the second delete is redundant here and elsewhere

Reply:
Yeah C++ won't consider the field type user-defined conversion if a viable template matches, and `const&` is viable for every type (even though const T&& and T&& shouldnt bind to it but have to for historic reasons :))
<img width="940" height="674" alt="image" src="https://github.com/user-attachments/assets/b16ba773-c931-4613-a4f1-4d24f221c146" />

Reply:
OK, updated